### DISCUSSION TITLE
Reference from child to supervisor

### DISCUSSION THREAD

**Discussion by dcadenas** [slawlor/ractor#213] [Q&A] [answered]
Hey! I have some doubt related to using `spawn_linked` for creating child actors. I'm already passing the supervisor's reference as the last argument for lifecycle and error monitoring. Now, I'm wondering, does this setup give the child any built-in way to ping back domain-specific messages to the supervisor, or am I missing something?

As it stands, I'm passing the supervisor's reference again to the child for those domain messages, essentially double-dipping on the supervisor reference in `spawn_linked`. It feels redundant. Is there a smarter way around this, or is the second pass just the way to go?

Is there a more elegant solution I'm overlooking? Thanks!

**Comment by slawlor** [comment, marked as answer]
It does not give a strongly typed message channel to the child from the parent. If you want that I'd suggest passing it as part of the child actor's startup arguments. 

There's too many generics at play, etc, in order to pass an `ActorRef` downstream into the non-strongly-typed logic at the core of `ractor`. 

Unfortunately there isn't a smarter way today. Unless you manually "link" the actors together (so start the child unlinked, then link it after the fact) however the risks any work specified in `post_start` causing an error on the child and not be captured in the supervision flow as a potential race condition. By using `spawn_linked`, it's guaranteed that an error in `pre_start` will immediately fail to "spawn" and any other errors will go through the supervision flow.
